packages/notion-utils/src/get-page-property.ts

```diff
@@ -56,21 +57,30 @@ export function getPageProperty(
           const property = block.properties[propertyId] as [['‣', [DateFormat]]]
           const formatDate = property[0][1][0][1]
 
-          if (formatDate.type == 'datetime') {
-            return new Date(
-              `${formatDate.start_date} ${formatDate.start_time}`
-            ).getTime()
-          } else if (formatDate.type == 'date') {
-            return new Date(formatDate.start_date).getTime()
-          } else if (formatDate.type == 'datetimerange') {
-            const { start_date, start_time, end_date, end_time } = formatDate
-            const startTime = new Date(`${start_date} ${start_time}`).getTime()
-            const endTime = new Date(`${end_date} ${end_time}`).getTime()
-            return [startTime, endTime]
-          } else {
-            const startTime = new Date(formatDate.start_date).getTime()
-            const endTime = new Date(formatDate.end_date).getTime()
-            return [startTime, endTime]
+          switch (formatDate.type) {
+            case 'date':
+            case 'datetime':
+              return getNotionDateTime(
```

Comment:
this use to return an integer (`getTime()`), but now these code paths are returning dates, which makes it hard to reason about the change in isolation